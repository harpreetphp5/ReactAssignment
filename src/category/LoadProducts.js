import React, { useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';
import { ProductsContext } from '../../src/helpers/ProductsContext';

const LoadProducts = () => {

    const [allProducts, cartItems, dispatch] = useContext(ProductsContext);

    const updateCart = (id) => {
        const matchedProd = allProducts.find(p => (p.title == id));

        const alreadyExisting = cartItems.find(p => (p.title == matchedProd.title));
        if (!!alreadyExisting) {
            dispatch({ type: 'UPDATE_CART', payload: { title: matchedProd.title, items: matchedProd, qty: 1 } });
        } else {
            dispatch({ type: 'ADD_CART', payload: { title: matchedProd.title, items: matchedProd, qty: 1 } });
        }
    }

    useEffect(() => {
        console.log(cartItems);
    }, [cartItems])

    return (

        allProducts.map(p => (
            <div key={p.title} className="productsGrid">
                <div className="imageWrap">
                    <img src={'./media/' + p.image} />
                    <div className="actionButtons">
                        <Link to={"/product/" + p.title} className="action">View Details</Link>
                        <button className="action" onClick={() => updateCart(p.title)} > Add to Cart</button>
                    </div>
                </div>
                <p className="brand">{p.brand}</p>
                <p className="title">{p.title}</p>
                <p className="price">${(p.price).toFixed(2)}</p>
            </div >
        ))

    )
}

export default LoadProducts;