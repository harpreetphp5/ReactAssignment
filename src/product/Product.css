.Product {
    width: 80%;
    margin: 50px auto;
    display: flex;
}
.firstHalf{
    width: 45%;
}
.firstHalf img{
    width: 100%;
}
.secondHalf{
    width: 39%;
    margin-left: 6%;
}
.actionProductBtn {
    background: #333;
    color: #fff !important;
    padding: 15px 35px;
    margin-bottom: 10px;
    font-size: 12px;
    float: right;
    border: none;
}
li.qtyTab {
    list-style: none;
    flex: 0.2;
    text-align: left;
    flex-direction: row;
    margin-top: 10px;
    height:0;
}
input.qtyType {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 25px;
    background: #aaa;
    border: none;
    padding: 15px;
}


div.qtyBox {
    float: left;
}

div.qtyBtns {
    width: 23px;
    height: 18px;
    display: flex;
    flex-direction: column;
}

div.qtyBtns button.second {
    margin-top: 2px
}
.productActionSection {
    width: 232px;
    margin: 0 auto;
}
