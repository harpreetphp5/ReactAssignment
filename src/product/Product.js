import React, { Component, useContext, useState, useEffect } from 'react';
import './Product.css';
import { ProductsContext, CartContext } from '../../src/helpers/ProductsContext';


const LoadProduct = ({ match }) => {
    const [allProducts, cartItems, dispatch] = useContext(ProductsContext);
    const [thisProd, setThisProd] = useState([]);
    const [qty, setQty] = useState(1);

    useEffect(() => {
        setQty(1);
        const newprod = allProducts.find(p => (p.title == match.params.id));
        setThisProd(newprod);

    }, []);

    const convertPrice = (price) => {
        return parseFloat(price).toFixed(2);
    }
    const UpdateQty = (identfier) => {
        if (qty == 0 && identfier == -1) {
            return;
        }
        setQty((identfier + qty));
    }
    const addToCart = () => {
        if (qty < 1) {
            return;
        }
        const alreadyExisting = cartItems.find(p => (p.title == thisProd.title));
        if (!!alreadyExisting) {
            dispatch({ type: 'UPDATE_CART', payload: { title: thisProd.title, items: thisProd, qty: qty } });
        } else {
            dispatch({ type: 'ADD_CART', payload: { title: thisProd.title, items: thisProd, qty: qty } });
        }
    }

    return (
        <div className="Product" >
            <div className="firstHalf">
                <img src={'/media/' + thisProd.image} />
            </div>
            <div className="secondHalf">
                <p className="brand">{thisProd.brand}</p>
                <h1>{thisProd.title}</h1>
                <p className="price">${convertPrice(thisProd.price)}</p>
                <p className="desc">{thisProd.description}</p>
                <hr />
                <div className="productActionSection">
                    <li className="qtyTab">
                        <div className="qtyBox">
                            <input className="qtyType" type="text" value={qty} />
                        </div>
                        <div className="qtyBtns">
                            <button onClick={() => UpdateQty(+1)}>+</button>
                            <button onClick={() => UpdateQty(-1)} className="second">-</button>
                        </div>
                    </li>
                    <button className="actionProductBtn" onClick={() => addToCart()}>Add to Cart</button>
                </div>
            </div>
        </div >
    )
}
const Product = ({ match }) => {

    return (
        <LoadProduct match={match} />
        )
};

export default Product;
