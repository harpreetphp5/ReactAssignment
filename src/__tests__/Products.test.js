import LoadProducts from '../category/LoadProducts';
import Product from '../product/Product';

it('renders without crashing', () => {

    expect(<Product />).toMatchSnapshot();
});

it('Loads Individual Product', () => {
    const match = { params: { id: 'Blue Stripe Stoneware Plate' } }
    expect(Product(match)).toMatchSnapshot();
});