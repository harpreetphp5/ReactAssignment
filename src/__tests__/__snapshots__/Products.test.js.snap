// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads Individual Product 1`] = `<LoadProduct />`;

exports[`renders without crashing 1`] = `<Product />`;
